layouts/error.vue

```diff
@@ -1,64 +1,79 @@
 <template>
     <b-container class="container-full-height">
-        <b-row class="min-vh-100" align-h="center" align-v="center">
-            <b-col class="text-center" md="9" lg="6">
+        <b-row align-h="center" align-v="center" class="min-vh-100">
+            <b-col class="text-center" lg="6" md="9">
                 <template v-if="is404">
                     <h1 class="font-weight-light">Page not found.</h1>
-                    <b-img src="/img/ohsnap.svg" fluid class="mt-5 mb-4"></b-img>
-                    <b-button variant="secondary" to="/"> go to dashboard </b-button>
+                    <b-img class="mt-5 mb-4" fluid src="/img/ohsnap.svg"></b-img>
+                    <b-button to="/" variant="secondary"> go to dashboard</b-button>
                 </template>
                 <template v-else>
                     <h1 class="font-weight-light">Oh snap!</h1>
                     <h5 class="text-grey font-weight-normal">Something is not right.</h5>
-                    <b-img src="/img/ohsnap.svg" fluid class="mt-5 mb-4"></b-img>
-                    <b-button variant="secondary" to="/"> go to dashboard </b-button>
+                    <b-img class="mt-5 mb-4" fluid src="/img/ohsnap.svg"></b-img>
+                    <b-button to="/" variant="secondary"> go to dashboard</b-button>
                 </template>
             </b-col>
         </b-row>
     </b-container>
 </template>
 
 <script lang="ts">
-import { Component, mixins, Prop } from 'nuxt-property-decorator'
-import BaseMixin from '~/mixins/BaseMixin'
+import { defineComponent, onMounted, useMeta, useRouter } from '@nuxtjs/composition-api'
+import { computed } from 'vue'
 
-@Component
-export default class NuxtError extends mixins(BaseMixin) {
-    @Prop(Object) error!: any
+export default defineComponent({
+    props: {
+        error: {
+            type: Object,
+            required: true,
+        },
+    },
+    setup(props) {
+        const { replace } = useRouter()
 
-    get statusCode() {
-        return (this.error && this.error.statusCode) || 500
-    }
+        const statusCode = computed(() => {
+            return (props.error && props.error.statusCode) || 500
+        })
 
-    get is404() {
-        return this.statusCode === 404
-    }
+        const is404 = computed(() => {
+            return statusCode.value === 404
+        })
 
-    head() {
-        return {
-            title: this.statusCode === 404 ? 'Page Not Found' : 'Oh snap!',
+        onMounted(() => {
+            switch (statusCode.value) {
+                case 401:
+                    replace('/login')
```

Comment:
Any reason why not router.push?

Reply:
didnt change logic just a simple migration to vue 3